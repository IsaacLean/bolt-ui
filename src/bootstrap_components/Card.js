/*
 * BOOTSTRAP CARD COMPONENT
 * https://v4-alpha.getbootstrap.com/components/card
 */

import React from 'react';

const Card = (props) => {
  let header;
  if(props.header) {
    header = <div className="card-header">
      {props.header}
    </div>;
  }

  let footer;
  if(props.footer) {
    footer = <div className="card-footer">
      {props.footer}
    </div>;
  }

  return <div className="card">
    {header}
    <div className="card-block">
      {props.content}
    </div>
    {footer}
  </div>;
};

export default Card;
