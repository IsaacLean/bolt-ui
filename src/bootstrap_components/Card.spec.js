import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Card from './Card';

test('Empty Card snapshot', () => {
  const component = shallow(<Card />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Card with header snapshot', () => {
  const component = shallow(<Card header="Header Text" />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Card with footer snapshot', () => {
  const component = shallow(<Card footer="Footer Text" />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
