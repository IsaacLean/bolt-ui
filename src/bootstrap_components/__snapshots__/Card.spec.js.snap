// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card with footer snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-block"
  />
  <div
    className="card-footer"
  >
    Footer Text
  </div>
</div>
`;

exports[`Card with header snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    Header Text
  </div>
  <div
    className="card-block"
  />
</div>
`;

exports[`Empty Card snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-block"
  />
</div>
`;
